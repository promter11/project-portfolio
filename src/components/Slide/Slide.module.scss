@import "../../styles/_variables";
@import "../../styles/_mixins";

.slide {
  position: relative;
  border-radius: calc(1.6rem + 0.2rem);
  box-shadow: 0 0 1rem 1rem rgba(0, 0, 0, 0.1);

  .header {
    .date {
      position: absolute;
      top: 2rem;
      right: 2rem;
      z-index: 5;
      @include text(1.2rem, 100, 1, none);
      color: $white-color;
    }

    .image {
      max-height: 36rem;
      @include width-height(100%, 100%);
      border-radius: 1.6rem 1.6rem 0 0;
      object-fit: cover;
      filter: brightness(0.5);
    }
  }

  .content {
    position: relative;
    padding: 3rem;
    border-radius: 0 0 1.6rem 1.6rem;
    background-color: rgba(0, 0, 0, 0.5);
    color: $white-color;

    .title {
      margin-bottom: 0.5rem;
      text-transform: uppercase;
    }

    .description {
      font-size: 1.4rem;
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;
    }

    .wrapper {
      position: absolute;
      top: -2rem;
      right: 2rem;
      display: flex;

      .button,
      .link {
        background-color: transparent;

        &:not(:last-child) {
          margin-right: 2rem;
        }

        .icon {
          @include width-height(4rem, 4rem);

          path {
            transition: fill 0.2s ease-in-out;
          }

          &:hover {
            path {
              fill: $social-fill-color;
            }
          }
        }
      }
    }
  }
}
