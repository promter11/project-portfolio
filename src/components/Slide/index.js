import React, { createElement, useState } from "react";
import { string, arrayOf, shape } from "prop-types";

import Heading from "../Heading";
import Modal from "../Modal";

import { ReactComponent as Info } from "../../assets/icons/portfolio/info.svg";
import styles from "./Slide.module.scss";
import modalStyles from "../Modal/Modal.module.scss";

const Slide = ({
  title,
  date,
  description,
  fullDescription,
  technologies,
  image,
  link: { text, path },
}) => {
  const [modalStatus, setModalStatus] = useState(false);

  return (
    <div className={styles.slide}>
      <div className={styles.header}>
        <span className={styles.date}>{date}</span>
        <img className={styles.image} src={image} alt="Preview" />
      </div>
      <div className={styles.content}>
        <Heading className={styles.title} level={2}>
          {title}
        </Heading>
        <p className={styles.description}>{description}</p>
        <div className={styles.wrapper}>
          <button
            className={styles.button}
            onClick={() => setModalStatus((value) => !value)}
          >
            {createElement(Info, { className: styles.icon }, null)}
          </button>
        </div>
      </div>
      <Modal
        modalStatus={modalStatus}
        toggleModal={() => setModalStatus((value) => !value)}
      >
        <div className={modalStyles.modalWrapper}>
          <div className={modalStyles.modalBlock}>
            <img className={modalStyles.modalImage} src={image} alt="Project" />
          </div>
          <div className={modalStyles.modalBlock}>
            <Heading className={modalStyles.modalTitle} level={3}>
              {title}
            </Heading>
            <p className={modalStyles.modalDesc}>{fullDescription}</p>
            <ul className={modalStyles.modalList}>
              {technologies.map((technology, index) => {
                return (
                  <li key={index} className={modalStyles.modalListItem}>
                    {technology}
                  </li>
                );
              })}
            </ul>
            <a className={modalStyles.modalLink} href={path} target="_blank">
              {text}
            </a>
          </div>
        </div>
      </Modal>
    </div>
  );
};

Slide.propTypes = {
  title: string.isRequired,
  date: string.isRequired,
  description: string.isRequired,
  fullDescription: string.isRequired,
  technologies: arrayOf(string).isRequired,
  image: string.isRequired,
  link: shape({
    text: string,
    path: string,
  }).isRequired,
};

export default Slide;
