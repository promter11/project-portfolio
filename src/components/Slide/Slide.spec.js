import React from "react";
import { shallow } from "enzyme";

import Slide from "./index";

const props = {
  title: "Title",
  date: "02 November, 1997",
  description: "Short description",
  fullDescription: "Full description",
  technologies: ["1", "2"],
  image: "../images/1",
  link: {
    text: "Link",
    path: "/",
  },
};

describe("Slide component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Slide {...props} />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("check setModalStatus function", () => {
    it("calls function in button", () => {
      wrapper.find(".button").simulate("click");

      expect(wrapper.find("Modal").prop("modalStatus")).toEqual(true);
    });

    it("calls function in Modal", () => {
      wrapper.find("Modal").props().toggleModal();

      expect(wrapper.find("Modal").prop("modalStatus")).toEqual(true);
    });
  });
});
