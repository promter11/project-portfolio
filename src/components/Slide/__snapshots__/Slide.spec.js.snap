// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide component renders correctly 1`] = `
<div
  className="slide"
>
  <div
    className="header"
  >
    <span
      className="date"
    >
      02 November, 1997
    </span>
    <img
      alt="Preview"
      className="image"
      src="../images/1"
    />
  </div>
  <div
    className="content"
  >
    <Heading
      className="title"
      level={2}
    >
      Title
    </Heading>
    <p
      className="description"
    >
      Short description
    </p>
    <div
      className="wrapper"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        <ForwardRef(SvgInfo)
          className="icon"
        />
      </button>
    </div>
  </div>
  <Modal
    modalStatus={false}
    toggleModal={[Function]}
  >
    <div
      className="modalWrapper"
    >
      <div
        className="modalBlock"
      >
        <img
          alt="Project"
          className="modalImage"
          src="../images/1"
        />
      </div>
      <div
        className="modalBlock"
      >
        <Heading
          className="modalTitle"
          level={3}
        >
          Title
        </Heading>
        <p
          className="modalDesc"
        >
          Full description
        </p>
        <ul
          className="modalList"
        >
          <li
            className="modalListItem"
            key="0"
          >
            1
          </li>
          <li
            className="modalListItem"
            key="1"
          >
            2
          </li>
        </ul>
        <a
          className="modalLink"
          href="/"
        >
          Link
        </a>
      </div>
    </div>
  </Modal>
</div>
`;
